src/contrib/cluster/Akka.Cluster.Metrics/Serialization/StandardMetrics.cs

```diff
@@ -0,0 +1,204 @@
+// //-----------------------------------------------------------------------
+// // <copyright file="StandardMetrics.cs" company="Akka.NET Project">
+// //     Copyright (C) 2009-2019 Lightbend Inc. <http://www.lightbend.com>
+// //     Copyright (C) 2013-2019 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// // </copyright>
+// //-----------------------------------------------------------------------
+
+using System;
+using Akka.Util;
+
+namespace Akka.Cluster.Metrics.Serialization
+{
+    /// <summary>
+    /// Definitions of the built-in standard metrics.
+    ///
+    /// The following extractors and data structures makes it easy to consume the <see cref="NodeMetrics"/>
+    /// in for example load balancers.
+    /// </summary>
+    public static class StandardMetrics
+    {
+        // Constants for the heap related Metric names
+        public const string HeapMemoryUsed = "heap-memory-used";
```

Comment:
Some ways of collecting this information in .NET:

1. The garbage collector: https://github.com/petabridge/NBench/blob/dev/src/NBench/Collection/Memory/GcTotalMemoryCollector.cs#L34
2. Performance Counters (Windows only)
3. Calling into native APIs via P/Invoke - would need different implementations for Windows vs. POSIX

Reply:
There might also be some new .NET Core instrumentation that is cross-platform. That's been on the to-do list for CoreCLR for some time now but I think they've made some progress against that since 2.2+

Reply:
Hmm, indeed - I will need to address this once will work on .NET metrics collector here, in this PR. I totally agree that we should look for something like _total allocated memory_, and will update collector's name (and metric's name as well) once will get clear understanding of what we are collecting at the end.

Reply:
Thanks for hints about research directions, by the way - and did not know anything about JVM heap :)